Mathlib/Topology/Algebra/LinearTopology.lean

```diff
@@ -261,6 +271,96 @@ theorem _root_.IsCentralScalar.isLinearTopology_iff [Module Rᵐᵒᵖ M] [IsCen
   · exact mk_of_hasBasis' Rᵐᵒᵖ (IsLinearTopology.hasBasis_submodule R)
       fun S r m hm ↦ unop_smul_eq_smul r m ▸ S.smul_mem _ hm
 
+variable (R M) in
+lemma nonarchimedeanAddGroup [IsTopologicalAddGroup M] [IsLinearTopology R M] :
+    NonarchimedeanAddGroup M := by
+  refine ⟨fun U hU ↦ ?_⟩
+  obtain ⟨N, hN, hNU⟩ := (hasBasis_submodule R).mem_iff.mp hU
+  exact ⟨⟨N.toAddSubgroup, N.toAddSubgroup.isOpen_of_mem_nhds hN⟩, hNU⟩
+
+lemma _root_.isLinearTopology_int_iff_nonarchimedeanAddGroup [IsTopologicalAddGroup M] :
+    IsLinearTopology ℤ M ↔ NonarchimedeanAddGroup M := by
+  refine ⟨fun _ ↦ IsLinearTopology.nonarchimedeanAddGroup ℤ M, fun _ ↦ ?_⟩
+  refine isLinearTopology_iff_exists_submodule_isOpen_subset.mpr fun s hs h0s ↦ ?_
+  obtain ⟨V, hV⟩ := NonarchimedeanAddGroup.is_nonarchimedean s (hs.mem_nhds h0s)
+  exact ⟨V.1.toIntSubmodule, V.2, hV⟩
+
+instance [NonarchimedeanAddGroup M] : IsLinearTopology ℤ M :=
```

Comment:
I can't decide whether this or that instance is useful/will add noise to the typeclass inference system. But having two equivalent types that are phrased differently looks annoying to me.

Reply:
I think this duplication is essentially unavoidable, and I believe the instance is essentially harmless (note there are no possible loops because `IsLinearTopology` --> `NonarchimedeanAddGroup` cannot be an instance).